src/dmangle.d

```diff
@@ -88,6 +94,9 @@ private immutable char[TMAX] mangleChar =
     //              X   // variadic T t...)
     //              Y   // variadic T t,...)
     //              Z   // not variadic, end of parameters
+    //              #   // Type backward reference
```

Comment:
the backrefernce can distinguish itself
no need to add so many prefixes I think.